internal/controller/streams.go

```diff
@@ -505,7 +509,30 @@ func ConnectStreams(ctx context.Context, logger *zap.SugaredLogger, envMap Envir
 				logger:             logger,
 				bufferedGrpcSyncer: bufferedGrpcSyncer,
 			}
-
+			ciliumFlowCollector, err := newCiliumFlowCollector(ctx, logger, envMap.CiliumNamespace)
+			falcoPresent := sm.isFalcoInstalled(ctx)
+			switch {
+			case ciliumFlowCollector != nil && falcoPresent:
+				streamClient.cniStatus = pb.CniPluginStatus_CNI_PLUGIN_STATUS_BOTH_PRESENT
+			case ciliumFlowCollector != nil:
+				streamClient.cniStatus = pb.CniPluginStatus_CNI_PLUGIN_STATUS_CILIUM_PRESENT
+			case falcoPresent:
+				streamClient.cniStatus = pb.CniPluginStatus_CNI_PLUGIN_STATUS_FALCO_PRESENT
+			default:
+				streamClient.cniStatus = pb.CniPluginStatus_CNI_PLUGIN_STATUS_NONE_FOUND
+				// Log cloudOperatorNoCNIPlugin
+				logEntry := &pb.LogEntry{
+					JsonMessage: "cloud-operator no CNI plugin found",
+				}
+				err := bufferedGrpcSyncer.client.Send(&pb.SendLogsRequest{
+					Request: &pb.SendLogsRequest_LogEntry{LogEntry: logEntry},
+				})
+				if err != nil {
+					logger.Warnw("Failed to send cloudOperatorNoCNIPlugin log entry", "error", err)
+				} else {
+					logger.Infow("Emitted cloudOperatorNoCNIPlugin log entry")
+				}
```

Comment:
There is no need to log.
Reporting this in `KubernetesClusterMetadata` is sufficient.
```suggestion
```